pointerToSlice/pointerToSlice.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package pointerToSlice
+
+import (
+	"go/ast"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "pointerToSlice",
+	Doc:  "check for usage of pointer to slice in function definitions",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	checkNode := func(expr ast.Expr) {
+		sexpr, ok := expr.(*ast.StarExpr)
+		if !ok {
+			return
+		}
+		if _, ok := sexpr.X.(*ast.ArrayType); ok {
+			pass.Reportf(sexpr.Pos(), "use of pointer to slice in function definition")
+			return
+		}
+		if tv, ok := pass.TypesInfo.Types[sexpr.X]; ok {
+			if strings.HasPrefix(tv.Type.String(), "[]") || strings.HasPrefix(tv.Type.Underlying().String(), "[]") {
```

Comment:
Hmm .. does this only check for pointers to slices or just plain slices also?

Reply:
We only check for plain slices here, given we are already in the "pointer to some type" case.